File: LeetCode/0406_Queue_Reconstruction_by_Height.js
/*
406. Queue Reconstruction by Height

https://leetcode.com/problems/queue-reconstruction-by-height/
*/

/**
 * @param {number[][]} people
 * @return {number[][]}
 */
var reconstructQueue = function (people) {
  people.sort((a, b) => {
    if (a[0] - b[0] > 0) return -1;
    else if (a[0] - b[0] < 0) return 1;
    else if (a[1] - b[1] > 0) return 1;
    else return -1;
  });
  let res = [];
  for (p of people) {
    res.splice(p[1], 0, p);
  }
  return res;
};

File: LeetCode/0406_Queue_Reconstruction_by_Height.py
"""
406. Queue Reconstruction by Height

https://leetcode.com/problems/queue-reconstruction-by-height/
"""
from typing import *


class Solution:
    def reconstructQueue(self, people: List[List[int]]) -> List[List[int]]:
        people.sort(key=lambda a: (-a[0], a[1]))
        res: List = []
        for p in people:
            res.insert(p[1], p)
        return res
